src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size;
 
+    public MateStack() {
+        stack = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == stack.length) {
+            grow();
+        }
+        stack[size++] = value;
     }
 
     public T peek() {
-        return null;
+        checkIfEmpty();
+        return stack[size - 1];
     }
 
     public T pop() {
-        return null;
+        checkIfEmpty();
+        T value = stack[size - 1];
+        stack[size-- - 1] = null;
```

Comment:
```suggestion
        stack[--size] = null;
```
first we decrease size by 1, then we take stack[new size which is - 1 already]